table/block_based_table_reader.cc

```diff
@@ -2406,6 +2432,29 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
           break;
         }
 
+        biter.Seek(key);
+
+        if (!biter.status().IsNotSupported()) {
+          // HashSeek supported
+
+          if (biter.Valid()) { // found
+            ParsedInternalKey parsed_key;
+            s = Status::OK();
+
+            if (!ParseInternalKey(biter.key(), &parsed_key)) {
+              s = Status::Corruption(Slice());
+            }
+
+            get_context->SaveValue(
+                parsed_key, biter.value(), &matched,
+                biter.IsValuePinned() ? &biter : nullptr);
+          }
+          break;
```

Comment:
If you move all the logic into iterator. It should be done in the same loop below, rather than creating several new lines of code here.